### PR TITLE
[VET-768] Support edge properties in SMS mappings (plus tests)

### PR DESCRIPTION
**Pull request by jmrog**
Adds support for both long-hand (`<< :some :triple :here >> a :triple`) and short-hand (`:some :triple { a :triple } :here`) edge property syntax in SMS. Also adds tests for these cases (some from docs, some from Stardog itself, and some made up). Thanks @alex-r-bigelow for working with me on this.

### CODE REVIEW

**Review by alex-r-bigelow — APPROVED** [review]
Probably should get an opinion beyond mine since we worked on this together, but FWIW, LGTM

File: src/__tests__/sms/fixtures.ts [hunk 1]
```diff
@@ -337,4 +337,84 @@ export const fixtures = {
...
+    select * from employees
+  }
+  TO {
+    ?emp a emp:Employee .
+
+    ?emp emp:firstName {
+      emp:since "?birth_date"^^xsd:date ;
+      emp:justAnotherFirstNameEdgeProp "?hire_date"^^xsd:date ;
+      emp:beginsAt "?birth_date"^^xsd:date
+    } ?first_name .
```
Comment (joshhk72) [inline src/__tests__/sms/fixtures.ts:412]: Maybe I'm doing something wrong, but I'm unable to save anything like this to a VG right now (I can save random edge properties, though).

**Review by joshhk72 — COMMENTED** [review]
One comment, since I'm not sure if it's actually a concern.

File: src/__tests__/sms/fixtures.ts (hunk 1 above)
Comment (jmrog) [reply src/__tests__/sms/fixtures.ts:412]: I just checked and was able to stuff like this without an issue. Maybe it was specific to what you were doing?

In any case, this is valid syntax (whether it is currently working when submitted) and is a test case used in Stardog's own tests. So, given that it works for me and _ought_ to work either way, I don't think it should block merging here. Maybe we can follow-up on your specific case to figure out what the issue is.

**Review by joshhk72 — APPROVED** [review]
LGTM!

**Pull request merged**